wswon/AndroidArchitectureStudy/app/src/main/java/com/architecture/study/view/coin/CoinListActivity.kt

```diff
@@ -29,54 +28,57 @@ class CoinListActivity : AppCompatActivity(), CoinListActivityContract.View {
         R.string.monetary_unit_4
     )
 
-    override fun onResume() {
-        super.onResume()
-        presenter.start()
-    }
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
 
-        presenter = CoinListActivityPresenter(
-            CoinRepositoryImp.getInstance(),
-            this@CoinListActivity,
-            false
-        )
+        coinListViewModel = CoinListViewModel(
+            CoinRepositoryImp.getInstance(Injection.provideCoinRemoteDataSource()),
+            tabList.map { getString(it) }
+        ).apply {
+            if(isConnectedApi){
+                getMarketList {
+                    if (it == CoinListViewModel.SUCCESS) {
+                        coinListViewModel.marketList.get()?.let { marketList->
+                            setTabPager(marketList)
+                        }
+                    } else {
+                        showMessage(it)
+                    }
+                }
+            }
+        }
     }
 
-    override fun showMessage(message: String) {
+    private fun showMessage(message: String) {
         Toast.makeText(this@CoinListActivity, message, Toast.LENGTH_SHORT).show()
     }
 
     /* tab layout && view pager init*/
-    override fun setTabPager(marketList: List<MarketResponse>) {
-        binding.tabLayoutMonetaryUnit.run {
-            tabList.map {
-                addTab(newTab().setText(getString(it)))
+    private fun setTabPager(marketList: List<MarketResponse>) {
+        binding.tabLayoutMonetaryUnit.setupWithViewPager(binding.viewPagerCoinList)
+
+        binding.viewPagerCoinList.run {
+            adapter = object : FragmentStatePagerAdapter(
+                supportFragmentManager,
+                BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
+            ) {
+
+                override fun getItem(position: Int): Fragment =
+                    CoinListFragment.newInstance(
+                        marketList
```

Comment:
데이터가 많은 경우, 시퀀스로 변환하여 사용하시는 것을 권장합니다.

Reply:
127dada 시퀀스로 변환한다는게 이게 맞는지 확인 부탁드립니다

Reply:
asSequence를 사용하여 변환해주시면 됩니다.
```suggestion
                        marketList.asSequence()
```

Reply:
408841e asSequece를 사용하여 처리 해보았습니다.